comm.cpp

```diff
@@ -215,6 +252,11 @@ int Endpoint::read_msg(struct buffer *pbuf)
     pbuf->len = expected_size;
 
     return 1;
+
+invalid_msg:
+    rx_buf.len -= expected_size;
+    memmove(rx_buf.data, rx_buf.data + expected_size, rx_buf.len);
```

Comment:
It is not discarding the entire buffer just the message(expected_size), but this block can be removed and use the _last_packet_len logic to do the same job.

Ok moving to a new method.